bug: <WasmEdge/example/plugin/get-string can not be compiled>

### Summary

Both when I try to build and run get-string plugin by cmake and running .wasm file , I will meet some error and `test.wasm` can not be compiled

### Current State

_No response_

### Expected State

_No response_

### Reproduction steps

1. If I run `mkdir build && cd build && cmake ..`, the terminal returns :
```
CMake Error at CMakeLists.txt:14 (target_include_directories):
  Error evaluating generator expression:

    $<TARGET_PROPERTY:wasmedgePlugin,INCLUDE_DIRECTORIES>

  Target "wasmedgePlugin" not found.

```

2. If I run `wat2wasm test.wat -o test.wasm && wasmedge test.wasm`, the error message is :
```
[2024-04-02 15:20:42.821] [error] instantiation failed: unknown import, Code: 0x302
[2024-04-02 15:20:42.822] [error]     When linking module: "module_wasm_name" , function name: "get_string"
[2024-04-02 15:20:42.822] [error]     At AST node: import description
[2024-04-02 15:20:42.822] [error]     At AST node: import section
[2024-04-02 15:20:42.823] [error]     At AST node: module

```

I asked this question in Discord. The admin helped me a lot and suggest me to raise an issue here.

### Screenshots

![image](https://github.com/WasmEdge/WasmEdge/assets/109454651/ef775273-3e3c-4fbb-852f-9179dfb35fce)
![image](https://github.com/WasmEdge/WasmEdge/assets/109454651/60660c7a-c522-4cf7-ba00-5c4a13d34bfc)


### Any logs you want to share for showing the specific issue

_No response_

### Components

CLI

### WasmEdge Version or Commit you used

commit ecdee81f71bd4ae980df609ebff7ac1a7bc2148e 

### Operating system information

Ubuntu 22.094

### Hardware Architecture

x86_64

### Compiler flags and options

_No response_

The build command should be:

```
cd WasmEdge
cmake -Bbuild -DWASMEDGE_BUILD_EXAMPLE=ON .
cmake --build build
```

And we will encounter another error due to the plugin API changes:

```
/Users/hydai/workspace/WasmEdge/examples/plugin/get-string/getstring.cpp:97:9: error: no type named 'PluginRegister' in namespace 'WasmEdge::Plugin'
Plugin::PluginRegister Register(&Descriptor);
~~~~~~~~^
1 error generated.
```

--- a/examples/plugin/get-string/getstring.cpp
+++ b/examples/plugin/get-string/getstring.cpp
@@ -78,22 +78,22 @@ create(const Plugin::PluginModule::ModuleDescriptor *) noexcept {
 }
 
 Plugin::Plugin::PluginDescriptor Descriptor{
-    .Name = "plugin_name",
-    .Description = "Example plugin",
-    .APIVersion = Plugin::Plugin::CurrentAPIVersion,
-    .Version = {0, 13, 4, 0},
-    .ModuleCount = 1,
-    .ModuleDescriptions =
-        (Plugin::PluginModule::ModuleDescriptor[]){
-            {
-                .Name = "module_name",
-                .Description = "Example module",
-                .Create = create,
-            },
+    /* Name */ "plugin_name",
+    /* Description */ "Example plugin",
+    /* APIVersion */ Plugin::Plugin::CurrentAPIVersion,
+    /* Version */ {0, 13, 5, 0},
+    /* ModuleCount */ 1,
+    /* ModuleDescriptions */
+    (Plugin::PluginModule::ModuleDescriptor[]){
+        {
+            /* Name */ "module_name",
+            /* Description */ "Example module",
+            /* Create */ create,
         },
-    .AddOptions = addOptions,
+    },
+    /* AddOptions */ addOptions,
 };
 
-Plugin::PluginRegister Register(&Descriptor);
+EXPORT_GET_DESCRIPTOR(Descriptor)
 
 } // namespace

